

<!DOCTYPE html>
<html lang="en">
<head>
  


<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Multiomics Integration Via Graph Learning [ Minsi Lu ]</title>

  <link rel="icon" href="/img/my.ico">


    <meta name="author" content="Minsi Lu">





  <link rel="alternate" href="/atom.xml " title="Minsi Lu" type="application/atom+xml">


  
    <link rel="stylesheet" href="/css/main.css">
  


<script type="text/javascript" async
 src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/MathJax.js?config=TeX-MML-AM_CHTML">
</script>

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [['$', '$'], ['\\(', '\\)']],
      processEscapes: true
    }
  });
  </script>
  
<meta name="generator" content="Hexo 7.1.1"></head>
  <body data-theme="light" class="notransition">
    <script>
      const body = document.body;
      const data = body.getAttribute("data-theme");
      const initTheme = (state) => {
        if (state === "dark") {
          body.setAttribute("data-theme", "dark");
        } else if (state === "light") {
          body.removeAttribute("data-theme");
        } else {
          localStorage.setItem("theme", data);
        }
      };
   
      initTheme(localStorage.getItem("theme"));
      
      setTimeout(() => body.classList.remove("notransition"), 75);
    </script>
  <div class="navbar" role="navigation">
    <nav class="menu">
      <input type="checkbox" id="menu-trigger" class="menu-trigger" />
      <label for="menu-trigger">
        <span class="menu-icon">
          <svg
            xmlns="http://www.w3.org/2000/svg"
            width="24"
            height="24"
            viewBox="0 0 512 512"
          >
            <path
              d="M64,384H448V341.33H64Zm0-106.67H448V234.67H64ZM64,128v42.67H448V128Z"
            />
          </svg>
        </span>
      </label>
      <a id="mode">
        <svg
          class="mode-sunny"
          xmlns="http://www.w3.org/2000/svg"
          width="24"
          height="24"
          viewBox="0 0 512 512"
        >
          <title>LIGHT</title>
          <line
            x1="256"
            y1="48"
            x2="256"
            y2="96"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="256"
            y1="416"
            x2="256"
            y2="464"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="403.08"
            y1="108.92"
            x2="369.14"
            y2="142.86"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="142.86"
            y1="369.14"
            x2="108.92"
            y2="403.08"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="464"
            y1="256"
            x2="416"
            y2="256"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="96"
            y1="256"
            x2="48"
            y2="256"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="403.08"
            y1="403.08"
            x2="369.14"
            y2="369.14"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="142.86"
            y1="142.86"
            x2="108.92"
            y2="108.92"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <circle
            cx="256"
            cy="256"
            r="80"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
        </svg>
        <svg
          class="mode-moon"
          xmlns="http://www.w3.org/2000/svg"
          width="24"
          height="24"
          viewBox="0 0 512 512"
        >
          <title>DARK</title>
          <line
            x1="256"
            y1="48"
            x2="256"
            y2="96"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="256"
            y1="416"
            x2="256"
            y2="464"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="403.08"
            y1="108.92"
            x2="369.14"
            y2="142.86"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="142.86"
            y1="369.14"
            x2="108.92"
            y2="403.08"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="464"
            y1="256"
            x2="416"
            y2="256"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="96"
            y1="256"
            x2="48"
            y2="256"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="403.08"
            y1="403.08"
            x2="369.14"
            y2="369.14"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="142.86"
            y1="142.86"
            x2="108.92"
            y2="108.92"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <circle
            cx="256"
            cy="256"
            r="80"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
        </svg>
      </a>
      <div class="trigger">
        <div class="trigger-container">
          
            
            
            
            
            
            
            <a class="menu-link " href="/"> Home</a>
          
            
            
            
            
            
            
            <a class="menu-link " href="/about/"> About me</a>
          
            
            
            
            
            
            
            <a class="menu-link " href="/publications/"> Publications</a>
          
            
            
            
            
            
            
            <a class="menu-link " href="/projects/"> Projects</a>
          
            
            
            
            
            
            
            <a class="menu-link " href="/cv/mycv.pdf"> CV</a>
          
            
            
            
            
            
            
            <a class="menu-link " target="_blank" rel="noopener" href="https://github.com/minsilu"> GitHub</a>
          
            
            
            
            
            
            
            <a class="menu-link " href="/blog/"> Blog</a>
          
            
            
            
            
            
            
            <a class="menu-link " href="/contact/"> Contact</a>
          
        </div>
      </div>
    </nav>
  </div>
  
<div class="wrapper">
  <header class="header">
    <h1 class="header-title center">Multiomics Integration Via Graph Learning</h1>
  </header>
  <main class="page-content" aria-label="Content">
      <div class="post">
        <p>Wenda Chu, Minsi Lu, Botian Wang, Sihang Zeng, Shiyu Zhao</p>
<p>Repo: <a target="_blank" rel="noopener" href="https://github.com/minsilu/Multiomics-Integration-via-Graph-Learning">Multiomics Integration Via Graph Learning</a></p>
<h2 id="Introduction"><a href="#Introduction" class="headerlink" title="Introduction"></a>Introduction</h2><p>In the evolving landscape of biological research, the advent of advanced sequencing technologies has underscored the significance of multi-omics data to dissect complex biological phenomena comprehensively. By integrating diverse omics datasets, researchers can unravel the intricate interplay among biomolecules, offering a holistic view far beyond what single-omics analyses could achieve. This integration, however, presents its own set of challenges, notably in reconciling matched and unmatched data through various sophisticated methods including latent space inference and nonlinear manifold alignment. </p>
<p>In this project, we focus on the integration of scRNA-seq data and scATAC-seq data through graph learning. Our method is based on GLUE. In this report, we first give an overview of GLUE, and then we introduce our improvements.</p>
<h2 id="Method"><a href="#Method" class="headerlink" title="Method"></a>Method</h2><p>Our method is based on GLUE, which is shown in the figure below. GLUE uses matrix factorization to model<br>the single-cell matrix of each omic. Each matrix is factorized into two vectors, single-cell embedding $U_i$ and omics feature embedding $V_i$, with $i$ indicating the index of certain omics. The cell embeddings are learned directly from the single-cell sequencing matrix, while the feature embeddings are learned from a knowledge based guidance graph through a graph encoder. </p>
<p>The guidance graph is the key to multi-omics integration. In GLUE, the guidance graph is constructed based on the overlap of features on the chromatin. The nodes are thefeatures, e.g., genes or peaks. An edge between two nodes indicates that the corresponding chromatin fragments<br>of the gene and the peak overlap with each other. The graph encoder is a graph convolutional network(GCN) with a simple mean aggregator. We believe that a better guidance graph and a better graph encoder could improve the performance of multi-omics integration, so we refine the model from these two perspectives.</p>
<p><img src="/projects/glue/f2.png" alt="f2"></p>
<h3 id="Graph-Construction"><a href="#Graph-Construction" class="headerlink" title="Graph Construction"></a>Graph Construction</h3><p>In the original method, they used genomic overlap simply. So the regulatory relationships inferred using graphs are limited to promoters and gene regions. More information, like cis-regulatory interactions on the upstream and downstream of the gene. CREs are often<br>but not always upstream of the transcription site. CREs are non-coding sequences, but are often co-accessible with genes. They regulate gene transcription by binding TF, and the locationis TF-motif. It is difficult to find information on CREs only from RNA-seq data, but we can infer cellular TF-gene interactions by TF-motif localization. Therefore, it is necessary to expand both upstream and downstream regions of the gene. </p>
<p>Genes are generally in an open state near the transcription start site (TSS), and decay with distance. To simulate real chromatin contact probability, we decay the weight of edges according to the distance of the ATAC peak from the TSS.</p>
<p>We connect ATAC peaks to RNA in the extended region and use a power-law function to decay the weights. For example, take extending the upstream and downstream 150kb of TSS. First, Convert RNA and ATAC data to BED format, locate the TSS and promoters. Second, Build the graph using ’scglue.genomics.window graph’, the weight is<br>$$ \omega &#x3D; (\frac{d+1000}{1000})^{-0.75} $$<br>Finally, We can get the expanded graph with weights.<br><img src="/projects/glue/f3.png" alt="f3"><br>The central node is the gene, and the surrounding node is the ATAC peak.</p>
<h3 id="Graph-Encoder"><a href="#Graph-Encoder" class="headerlink" title="Graph Encoder"></a>Graph Encoder</h3><p>In this section, we introduce our method for aggregating information on the guidance graph, highlighting the limitations of the graph aggregator proposed by previous studies, and presenting a novel aggregation approach based on our observations.</p>
<p>In the original model of GLUE, the authors use GCN, a commonly used graph neural network, to encode the guidance graph. The following is a typical form for GNN,</p>
<p><img src="/projects/glue/e1.png" alt="e1"></p>
<p>$f$: the aggregator to aggregate neighborhood information into message,<br>$\phi$: the combinator to combine the updated the message from neighborhoods with the target node. As for GCN, $f$ is the MEAN aggregator, that is,<br><img src="/projects/glue/e2.png" alt="e2"></p>
<h4 id="Indistinguishability-in-Single-Aggregator"><a href="#Indistinguishability-in-Single-Aggregator" class="headerlink" title="Indistinguishability in Single Aggregator"></a>Indistinguishability in Single Aggregator</h4><p>The original GLUE model employs a Graph Convolutional Network (GCN) to encode the guidance graph, using a MEAN aggregator to combine information from neighboring nodes. However, this approach has limitations in expressiveness, losing structural information in encoded graphs. We illustrate this issue with an example where a MEAN aggregator cannot distinguish between two graphs representing different biological scenarios due to its inability to capture the unique contributions of individual nodes.<br><img src="/projects/glue/f4.png" alt="f43"></p>
<h4 id="GCN-with-Principal-Neighborhood-Aggregation-PNA"><a href="#GCN-with-Principal-Neighborhood-Aggregation-PNA" class="headerlink" title="GCN with Principal Neighborhood Aggregation (PNA)"></a>GCN with Principal Neighborhood Aggregation (PNA)</h4><p>To address the limitations of a single aggregator, we propose the use of Principal Neighborhood Aggregation (PNA), which incorporates multiple aggregators in the graph encoding process. This method enhances the graph encoder’s ability to preserve biological information by utilizing a set of independent aggregators in each GCN layer, thereby achieving better distinguishability for guidance graphs.</p>
<p><img src="/projects/glue/f5.png" alt="f5"></p>
<p>The PNA approach is defined as follows:<br><img src="/projects/glue/e3.png" alt="e3"></p>
<p>where ${f_i, i \in I}$ represents a set of predefined aggregators, including Mean, Max, Min, and standard deviation. This diverse set of aggregators allows for a richer representation of the guidance graph’s structure, significantly improving the model’s ability to guide multi-omics integration with enhanced fidelity to biological information.<br><img src="/projects/glue/e4.png" alt="e4"></p>
<h2 id="Results"><a href="#Results" class="headerlink" title="Results"></a>Results</h2><p>We present the experimental results of our framework in two downstream applications: Cell Integration and Regulatory Inference, which are two typical applications for multiomics integration frameworks.</p>
<h3 id="Cell-Integration"><a href="#Cell-Integration" class="headerlink" title="Cell Integration"></a>Cell Integration</h3><p>Cell integration intends to make use of the cell encoders to embed the sequencing data from different cell types and omics into a shared latent space. Effective multiomics integration methods should simultaneously <strong>distinguish different cell types</strong> and <strong>align the distinctive omics layers</strong> (the scATAC-seq and scRNA-seq in our setting). We adopt six metrics used in GLUE to evaluate the performance of cell integration. </p>
<p>The metrics can be classified into two types: </p>
<p>(1) Biological conservation metrics that evaluate the resolution of clustered cell types, including Average Silhouette Width, Normalized Mutual Information, and Mean Average Precision. Higher scores will be assigned for clustering results that successfully distinguish the different cell types. </p>
<p>(2)<br>Omics mixing metrics that measure the extent that different omics layers are mixed together, including Graph Connectivity, Average Silhouette Width Batch, and Seurat Alignment Score. Higher scores will be given if the scATAC-seq and scRNA-seq are evenly distributed in the latent embedding space. For brevity, we omit the mathematical formulations of the metrics, which can be found in the GLUE paper.</p>
<p>We compare these metrics across three model variants:</p>
<ul>
<li>GLUE is the model presented in the original GLUE paper.</li>
<li>GLUE+PNA leverages multiple aggregators when encoding the guidance graph.</li>
<li>GLUE+PNA+Decay further modifies the guidance graph to incorporate more gene-ATAC peak relationships.<br><img src="/projects/glue/t1.png" alt="t1"></li>
</ul>
<p>The overall performance is summarized in Table 1. Firstly, despite that for all metrics except Seurat Alignment Score, the original GLUE model still achieves the best results, our methods still perform comparatively. We hypothesize that our efforts are targeted at the construction and encoding of the prior guidance graph, which shall naturally lead to improvements of the node features rather than the cell features. However, only cell features are utilized in the cell integration process. Hence, we cannot see significant improvements of the related metrics. We will discuss more benefits of our methods in the Regulatory Inference Section. Secondly,<br>when comparing the two variants of our model, we can see that GLUE+PNA+Decay outperforms GLUE+PNA<br>in all biological conservation metrics. It may be credited to the additional prior knowledge introduced by the augmented gene-ATAC graph, which contributes to the bettered biological soundness of the cell features.<br><img src="/projects/glue/f6.png" alt="f6"></p>
<h3 id="Regulatory-Inference"><a href="#Regulatory-Inference" class="headerlink" title="Regulatory Inference"></a>Regulatory Inference</h3><p>Another important application of the multiomics integration framework is integrative regulatory inference. Following the GLUE pipeline, we first extract a directed non-cyclic skeleton graph with high-variant genes and ATAC peaks as nodes from the constructed guidance graph, on which we mine the regulatory relationships.</p>
<p>Then, we use the trained embeddings of the gene and ATAC peak features as node features. We calculate the cosine similarity of the features to obtain the regulatory scores, which are compared to a null distribution and are each assigned a significant score. The edges receiving significant regulatory scores are extracted as valid regulation relationships.</p>
<p>We then conduct a qualitative analysis on the mined regulatory graph. Firstly, as shown in the table below. We notice that around 1100 more relationships are extracted in our regulatory graph. Moreover, we found 41 genes uniquely appearing in our regulation graph, including important protein encoding genes such as CtsW, Bola1 and Figla. Furthermore, we examined that the weights related to the 41 gene nodes are relatively low, which confirms that they are relevant to the decayed edges we constructed in the extended region of the guidance graph. These findings demonstrate the strengths of our methods in finding richer regulatory relationships compared to the vanilla GLUE framework. Note that we could also combine the extracted gene-peak edges with the public peak-TF Motif data to form gene-TF regulation network, which is of more biological significance.<br><img src="/projects/glue/t2.png" alt="t2"></p>

       </div>
         <div class="dis">
    
     

  </div>
  </main>
</div>
  <footer class="footer">
    <small class="footer_copyright">
      <div id="bottom-inner">
        © 2021 Site by Minsi Lu.
      </div>
    </small>
  </footer>
  
  

    
      <script src="/js/main.js"></script>
    
      <script src="/js/custom-filter.js"></script>
    
  
  <script>
    window.FPConfig = {
      delay: 0,
      ignoreKeywords: [],
      maxRPS: 3,
      hoverDelay: 50,
    };
  </script>
  <script defer src="https://cdn.jsdelivr.net/gh/gijo-varghese/flying-pages@2.1.2/flying-pages.min.js"></script>



  
</body>
</html>
