

<!DOCTYPE html>
<html lang="en">
<head>
  


<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Capricorn: a multi-view diffusion model for Hi-C contact matrix resolution enhancement [ Minsi Lu ]</title>

  <link rel="icon" href="/img/my.ico">


    <meta name="author" content="Minsi Lu">





  <link rel="alternate" href="/atom.xml " title="Minsi Lu" type="application/atom+xml">


  
    <link rel="stylesheet" href="/css/main.css">
  


<script type="text/javascript" async
 src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/MathJax.js?config=TeX-MML-AM_CHTML">
</script>

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [['$', '$'], ['\\(', '\\)']],
      processEscapes: true
    }
  });
  </script>
  
<meta name="generator" content="Hexo 7.1.1"></head>
  <body data-theme="light" class="notransition">
    <script>
      const body = document.body;
      const data = body.getAttribute("data-theme");
      const initTheme = (state) => {
        if (state === "dark") {
          body.setAttribute("data-theme", "dark");
        } else if (state === "light") {
          body.removeAttribute("data-theme");
        } else {
          localStorage.setItem("theme", data);
        }
      };
   
      initTheme(localStorage.getItem("theme"));
      
      setTimeout(() => body.classList.remove("notransition"), 75);
    </script>
  <div class="navbar" role="navigation">
    <nav class="menu">
      <input type="checkbox" id="menu-trigger" class="menu-trigger" />
      <label for="menu-trigger">
        <span class="menu-icon">
          <svg
            xmlns="http://www.w3.org/2000/svg"
            width="24"
            height="24"
            viewBox="0 0 512 512"
          >
            <path
              d="M64,384H448V341.33H64Zm0-106.67H448V234.67H64ZM64,128v42.67H448V128Z"
            />
          </svg>
        </span>
      </label>
      <a id="mode">
        <svg
          class="mode-sunny"
          xmlns="http://www.w3.org/2000/svg"
          width="24"
          height="24"
          viewBox="0 0 512 512"
        >
          <title>LIGHT</title>
          <line
            x1="256"
            y1="48"
            x2="256"
            y2="96"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="256"
            y1="416"
            x2="256"
            y2="464"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="403.08"
            y1="108.92"
            x2="369.14"
            y2="142.86"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="142.86"
            y1="369.14"
            x2="108.92"
            y2="403.08"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="464"
            y1="256"
            x2="416"
            y2="256"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="96"
            y1="256"
            x2="48"
            y2="256"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="403.08"
            y1="403.08"
            x2="369.14"
            y2="369.14"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="142.86"
            y1="142.86"
            x2="108.92"
            y2="108.92"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <circle
            cx="256"
            cy="256"
            r="80"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
        </svg>
        <svg
          class="mode-moon"
          xmlns="http://www.w3.org/2000/svg"
          width="24"
          height="24"
          viewBox="0 0 512 512"
        >
          <title>DARK</title>
          <line
            x1="256"
            y1="48"
            x2="256"
            y2="96"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="256"
            y1="416"
            x2="256"
            y2="464"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="403.08"
            y1="108.92"
            x2="369.14"
            y2="142.86"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="142.86"
            y1="369.14"
            x2="108.92"
            y2="403.08"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="464"
            y1="256"
            x2="416"
            y2="256"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="96"
            y1="256"
            x2="48"
            y2="256"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="403.08"
            y1="403.08"
            x2="369.14"
            y2="369.14"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="142.86"
            y1="142.86"
            x2="108.92"
            y2="108.92"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <circle
            cx="256"
            cy="256"
            r="80"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
        </svg>
      </a>
      <div class="trigger">
        <div class="trigger-container">
          
            
            
            
            
            
            
            <a class="menu-link " href="/"> Home</a>
          
            
            
            
            
            
            
            <a class="menu-link " href="/about/"> About me</a>
          
            
            
            
            
            
            
            <a class="menu-link " href="/publications/"> Publications</a>
          
            
            
            
            
            
            
            <a class="menu-link " href="/projects/"> Projects</a>
          
            
            
            
            
            
            
            <a class="menu-link " href="/cv/mycv.pdf"> CV</a>
          
            
            
            
            
            
            
            <a class="menu-link " target="_blank" rel="noopener" href="https://github.com/minsilu"> GitHub</a>
          
            
            
            
            
            
            
            <a class="menu-link " href="/blog/"> Blog</a>
          
            
            
            
            
            
            
            <a class="menu-link " href="/contact/"> Contact</a>
          
        </div>
      </div>
    </nav>
  </div>
  
<div class="wrapper">
  <header class="header">
    <h1 class="header-title center">Capricorn: a multi-view diffusion model for Hi-C contact matrix resolution enhancement</h1>
  </header>
  <main class="page-content" aria-label="Content">
      <div class="post">
        <p>Paper: <a target="_blank" rel="noopener" href="https://www.biorxiv.org/content/10.1101/2023.10.25.564065v1">Capricorn: a multi-view diffusion model for Hi-C contact matrix resolution enhancement</a></p>
<h2 id="Introduction"><a href="#Introduction" class="headerlink" title="Introduction"></a>Introduction</h2><p>High-throughput Chromosome Conformation Capture (Hi-C) technology has revolutionized our understanding of the 3D architecture of the genome.</p>
<p>However, a significant challenge in utilizing Hi-C data is its requirement for high-coverage sequencing to achieve the resolution necessary for detecting fine-scale chromatin structures, such as chromatin loops. High-coverage Hi-C datasets demand extensive sequencing efforts, making them costly and time-consuming to produce. </p>
<ul>
<li>According to Rao et al., 2014, a ideal resolution for hic matrix: 80% of loci have at least 1,000 contacts</li>
</ul>
<p>This limitation has prompted the development of computational methods aimed at enhancing the resolution of low-coverage Hi-C data, thereby enabling more detailed genomic analyses at reduced costs. </p>
<h2 id="Problem-Setting"><a href="#Problem-Setting" class="headerlink" title="Problem Setting"></a>Problem Setting</h2><p>For a set of experimental cell lines $C$, interaction frequency matrices are available for chromosomes $\Xi$. The dataset consists of pairs of low- and high-coverage contact matrices $ D &#x3D; \left { (X^{(c)}<em>{\xi}, Y^{(c)}</em>{\xi}) \right} $ for each cell line $c \in C$ and chromosome $\xi \in \Xi$, all at high resolution.</p>
<p>The goal is to develop a model $f$ that approximates $Y^{(c)}<em>{\xi} \approx f(X^{(c)}</em>{\xi})$ for all cell lines $c$ and chromosomes $\xi$. The model $f$ should also generalize well to new cell lines and different chromosomes.</p>
<h3 id="Constraints"><a href="#Constraints" class="headerlink" title="Constraints"></a>Constraints</h3><ul>
<li>The focus is on enhancing intrachromosomal contacts within a 2 Mb range.</li>
<li>Each chromosome $\xi$ has $N_{\xi}$ total base pairs.</li>
<li>Fixed resolution $\Delta$, resulting in contact matrix shapes of $N_{\xi}&#x2F;\Delta \times N_{\xi}&#x2F;\Delta$.</li>
<li>The high-coverage version $Y^{(c)}_{\xi}$ contains $\lambda$-fold more contacts than its low-coverage counterpart.</li>
</ul>
<h3 id="Matrix-Details"><a href="#Matrix-Details" class="headerlink" title="Matrix Details"></a>Matrix Details</h3><ul>
<li>Each entry $X^{(c)}_{\xi}[i, j]$ represents a $\Delta^2 &#x3D; 10kb \times 10kb$ region of genomic interactions between the genomic regions $[10i , \text{kb}, 10(i + 1) , \text{kb})$ and $[10j , \text{kb}, 10(j + 1) , \text{kb})$, with $\Delta &#x3D; 10 , \text{kb}$.</li>
<li>The matrices are tiled into $40 \times 40$ non-overlapping submatrices $X^{(c)}_{\xi}[i : i + 40, j : j + 40]$, each covering a $400^2 , \text{kb}$ region, aligning with existing Hi-C resolution enhancement frameworks.</li>
</ul>
<h2 id="Capricorn-Model-Overview"><a href="#Capricorn-Model-Overview" class="headerlink" title="Capricorn Model Overview"></a>Capricorn Model Overview</h2><p>Capricorn is a novel machine learning model designed to enhance the resolution of Hi-C contact matrices, focusing on the accurate detection and representation of chromatin loops. Its approach diverges from traditional image resolution enhancement techniques by incorporating biological insights into the model, leveraging a diffusion probability model as its backbone. This section delves into the technical aspects of Capricorn, explaining its methodology, the problem it addresses, and its innovative use of diffusion models for biological data enhancement.</p>
<h3 id="Methodology"><a href="#Methodology" class="headerlink" title="Methodology"></a>Methodology</h3><p>Capricorn addresses the challenge of enhancing low-coverage Hi-C data by integrating small-scale chromatin features (such as TADs and loops) as additional inputs, alongside the primary Hi-C contact matrix. This multi-view approach allows Capricorn to generate high-resolution contact matrices that more accurately reflect the underlying biological structures.</p>
<ol>
<li><p><strong>Input Data Preparation</strong>: The model takes a low-coverage Hi-C contact matrix and enriches it with derived features, including:</p>
<ul>
<li><strong>Distance-corrected matrices</strong>: Adjust for bias based on inter-locus distance, normalizing contacts based on expected distances to handle the diagonal dominance in the contact matrices. The expected matrix $E(X)$ is computed by averaging contacts along each diagonal, and the observed matrix is normalized accordingly.</li>
<li><strong>TAD scores</strong>: Utilize insulation scores (IS) for TAD detection, identifying insulated regions with low scores and within-TAD regions with high scores. The IS scores are smoothed over a 21-bin domain, and labels are assigned based on monotonically decreasing IS centered around some locus.</li>
<li><strong>Loop p-value and ratio</strong>: Follow the Hi-C Computational Unbiased Peak Search (HiCCUPS) algorithm for loop detection, assessing whether measured contacts are significantly more frequent than expected. This involves combining a 10x10 donut kernel with other kernels centered at a specific locus and computing the loop ratio and p-values based on a distance-based expected matrix.</li>
</ul>
</li>
<li><p><strong>Diffusion Model Backbone</strong>:<br>Capricorn leverages conditional diffusion probability models, treating the combined contact matrix and its derived chromatin feature views as a multi-channel image. We use the conditional diffusion probability model <a target="_blank" rel="noopener" href="https://imagen.research.google/">Imagen</a> as the resolution enhancement backbone model, updating the model to condition on low-coverage contact matrices rather than text. We choose Imagen rather than other image diffusion models due to its efficient U-Net architecture, which is faster and more memory efficient than other diffusion generators. This approach allows Capricorn to generate high-resolution contact matrices by effectively incorporating both the primary Hi-C matrix and additional biological views (TADs, loops, distance-normalized counts).</p>
</li>
<li><p><strong>Loss Function in Capricorn Model</strong>:In the Capricorn model, the inputs are derived directly from the low-coverage experimental matrix $X$, ensuring that there’s no prior knowledge of the high-coverage contact matrix $Y$. This approach maintains Capricorn’s utility in practical inference scenarios. The inputs and outputs are structured as follows:</p>
<ul>
<li>Input: $$\widetilde{X}(X) &#x3D; [X, X^{(oe)}, X^{(tad)}, X^{(loop-p)}, X^{(loop-r)}] \in \mathbb{R}^{5 	\times N&#x2F;\Delta 	\times N&#x2F;\Delta}$$</li>
<li>Output: $$\widetilde{Y}(Y) &#x3D; [Y, Y^{(oe)}, Y^{(tad)}, Y^{(loop-p)}, Y^{(loop-r)}] \in \mathbb{R}^{5 	\times N&#x2F;\Delta 	\times N&#x2F;\Delta}$$</li>
</ul>
<p>To avoid the dominance of any single experimental view during model training, an initial exploratory version of Capricorn is trained end-to-end. The validation set loss, $L’<em>{val}(\widehat{\widetilde{Y}} | \widetilde{X})$, is then assessed to evaluate the contribution of each view to the overall loss. This analysis leads to the creation of a weight vector $\omega \in \R</em>+^5$, designed to balance each view’s contribution to the loss function. Consequently, during the full model training, each channel is weighted by $\sqrt{\omega}$ to ensure an equitable contribution to the input and output, optimizing the model’s performance and utility.</p>
</li>
<li><p><strong>Training and Inference</strong>: During training, Capricorn learns to predict the high-resolution contact matrix from its low-resolution counterpart and the additional chromatin features. The model is trained to minimize the mean squared error between its predictions and the ground truth high-resolution matrices, with adjustments to ensure that all input views contribute equally to the loss.<br><img src="/projects/capricorn/f1.png" alt="training procedure"></p>
</li>
</ol>
<h3 id="Performance-Measures"><a href="#Performance-Measures" class="headerlink" title="Performance Measures"></a>Performance Measures</h3><p>Capricorn’s performance is evaluated using both image-based and biologically motivated metrics. The model’s predicted high-coverage contact matrix is compared against the true high-coverage matrix using mean squared error (MSE) and loop F1 score. The loop F1 score is calculated based on the number of correctly predicted loops, with a tolerance for positional discrepancies, to assess the model’s ability to accurately capture biologically relevant chromatin structures.</p>
<p>loop F1 score: $$ F1 &#x3D; \frac{TP}{TP+\frac{1}{2}(FP+FN) } $$ with a 5 pixel (50kb) tolerance range.<br>The model’s performance has been validated across various cell lines and chromosomes, showcasing its general applicability and potential for uncovering new insights into genome architecture.</p>
<h3 id="Hi-C-data-preprocessing"><a href="#Hi-C-data-preprocessing" class="headerlink" title="Hi-C data preprocessing"></a>Hi-C data preprocessing</h3><ol>
<li><strong>Dataset</strong>: GM12878 Epstein-Barrvirus-infected human lymphoblastoid cell line and K562 human chronic myelogenous leukemia lymphoblast cell lines from the Rao et al. (2014) dataset (accession code GSE63525)</li>
<li><strong>Preprocessing:</strong> Restricting the contact matrix to read mapping quality ≥ 30 and processed to 10 kilobase (kb) resolution, following previous work. We adopt the contact matrix preprocessing techniques from HiCARN and DeepHiC, including tiling the contact matrices into 40×40 submatrices, only retaining submatrices in the 2 megabase (Mb) region around the diagonal, clamping the high-coverage matrix to [0, 255] and then normalizing to [0, 1], and clamping thelow-coverage matrix to [0, 100] and then normalizing to [0, 1].</li>
<li><strong>Downsample:</strong> In order to simulate low-coverage data, we randomly downsampled the GM12878 and K562 cell line Hi-C matrices to 1&#x2F;16 of the original read count.</li>
<li>We train on the GM12878 data and test the model on K562; in the second, we train on K562 data and test on GM12878. In both experiments, we withhold chromosomes 4, 5, 11, and 14 from the training cell line as our validation set.</li>
</ol>
<h2 id="Results"><a href="#Results" class="headerlink" title="Results"></a>Results</h2><h3 id="Capricorn-accurately-enhances-contact-matrices-and-loop-features"><a href="#Capricorn-accurately-enhances-contact-matrices-and-loop-features" class="headerlink" title="Capricorn accurately enhances contact matrices and loop features"></a>Capricorn accurately enhances contact matrices and loop features</h3><p>Capricorn demonstrated remarkable performance in enhancing the resolution of Hi-C contact matrices and accurately identifying chromatin loops. This success was evident in cross-cell-line experiments, where Capricorn was trained on simulated low-coverage data from one cell line and tested on another. Notably, it surpassed other models in enhancing chromatin loops and accuracy in generating high-coverage data. For instance, in testing across GM12878 and K562 cell lines, Capricorn achieved an average loop F1-score improvement and significantly lower mean squared error (MSE) compared to other methods, indicating its precision in reflecting true genomic interactions.<br><img src="/projects/capricorn/f2.png" alt="r1"></p>
<h3 id="Small-scale-chromatin-features-are-critical-to-model-improvement-and-are-model-agnostic"><a href="#Small-scale-chromatin-features-are-critical-to-model-improvement-and-are-model-agnostic" class="headerlink" title="Small-scale chromatin features are critical to model improvement and are model-agnostic"></a>Small-scale chromatin features are critical to model improvement and are model-agnostic</h3><p>Further analysis revealed that incorporating small-scale chromatin features (e.g., TADs and loops) as additional inputs significantly and enhancing the small-scale chromatin features<br>in addition to the Hi-C matrices increased Capricorn’s performance. This approach not only improved the resolution enhancement task but also the model’s ability to identify structurally meaningful contacts from the enhanced matrices. </p>
<ol>
<li>Primary view: This setting uses the comparison models’ default resolution enhancement pipelines with the low- and high-coverage Hi-C matrices as input and output. </li>
<li>Five-view input only: This setting uses the additional chromatin feature views as input to the model, but is still trained to only predict the high-coverage view. </li>
<li>Full five-view model: This setting uses Capricorn’s complete multi-view setting, including all five chromatin feature views as input and training the model to enhance the small-scale chromatin features in addition to the Hi-C matrices.<br><img src="/projects/capricorn/f3.png" alt="r2"></li>
</ol>
<h3 id="Capricorn-generalizes-across-chromosomes"><a href="#Capricorn-generalizes-across-chromosomes" class="headerlink" title="Capricorn generalizes across chromosomes"></a>Capricorn generalizes across chromosomes</h3><p>A rigorous testing regime confirmed Capricorn’s ability to generalize its learning across different chromosomes and cell lines, underscoring its robustness. The model effectively transferred learned patterns to new genomic loci, outperforming comparison approaches in identifying accurate chromatin loops, further evidenced by cross-chromosome and intra-cell-line generalizations.</p>
<p><img src="/projects/capricorn/f4.png" alt="r3"></p>
<h2 id="Discussion-and-Future-Directions"><a href="#Discussion-and-Future-Directions" class="headerlink" title="Discussion and Future Directions"></a>Discussion and Future Directions</h2><p>Capricorn represents a significant advancement in the field of Hi-C resolution enhancement by explicitly modeling the biological underpinnings of contact matrices through the incorporation of small-scale chromatin features. This approach has not only demonstrated Capricorn’s superiority in enhancing Hi-C data resolution but also highlighted its applicability to a broader range of resolution enhancement problems beyond its initial scope.</p>
<h3 id="Future-Directions"><a href="#Future-Directions" class="headerlink" title="Future Directions"></a>Future Directions</h3><p>Looking forward, the Capricorn framework offers ample room for expansion to incorporate further biological views tailored to specific downstream applications. For instance, integrating structural information covering larger genomic loci could enhance the identification of TADs and A&#x2F;B compartments, thereby broadening the model’s applicability and utility in genomic research.</p>
<p>Additionally, exploring the impact of varying model backbones on the multi-view resolution enhancement problem presents an intriguing avenue for future research. This could involve assessing the model’s adaptability to different genomic datasets, including those from other species, thereby extending its utility to a cross-species transfer learning context.</p>
<p>Moreover, applying Capricorn to a wider array of Hi-C cell line data and investigating the effects of training data size could yield insights into optimizing the model’s performance. Furthermore, the potential to adapt Capricorn for use with other types of contact map data, such as micro-C, suggests opportunities for broadening the model’s applicability within genomic research.</p>
<p>Lastly, future studies might explore the performance of Capricorn across various loop calling methods, enhancing the model’s capacity to accurately identify biologically relevant chromatin structures. This would not only solidify Capricorn’s standing as a robust tool for Hi-C resolution enhancement but also contribute to our understanding of the complex architecture of the genome.</p>

       </div>
         <div class="dis">
    
     

  </div>
  </main>
</div>
  <footer class="footer">
    <small class="footer_copyright">
      <div id="bottom-inner">
        © 2021 Site by Minsi Lu.
      </div>
    </small>
  </footer>
  
  

    
      <script src="/js/main.js"></script>
    
      <script src="/js/custom-filter.js"></script>
    
  
  <script>
    window.FPConfig = {
      delay: 0,
      ignoreKeywords: [],
      maxRPS: 3,
      hoverDelay: 50,
    };
  </script>
  <script defer src="https://cdn.jsdelivr.net/gh/gijo-varghese/flying-pages@2.1.2/flying-pages.min.js"></script>



  
</body>
</html>
