

<!DOCTYPE html>
<html lang="en">
<head>
  


<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>PPI Network Guided Driver Target Discovery [ Minsi Lu ]</title>

  <link rel="icon" href="/img/my.ico">


    <meta name="author" content="Minsi Lu">





  <link rel="alternate" href="/atom.xml " title="Minsi Lu" type="application/atom+xml">


  
    <link rel="stylesheet" href="/css/main.css">
  


<script type="text/javascript" async
 src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/MathJax.js?config=TeX-MML-AM_CHTML">
</script>

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [['$', '$'], ['\\(', '\\)']],
      processEscapes: true
    }
  });
  </script>
  
<meta name="generator" content="Hexo 7.1.1"></head>
  <body data-theme="light" class="notransition">
    <script>
      const body = document.body;
      const data = body.getAttribute("data-theme");
      const initTheme = (state) => {
        if (state === "dark") {
          body.setAttribute("data-theme", "dark");
        } else if (state === "light") {
          body.removeAttribute("data-theme");
        } else {
          localStorage.setItem("theme", data);
        }
      };
   
      initTheme(localStorage.getItem("theme"));
      
      setTimeout(() => body.classList.remove("notransition"), 75);
    </script>
  <div class="navbar" role="navigation">
    <nav class="menu">
      <input type="checkbox" id="menu-trigger" class="menu-trigger" />
      <label for="menu-trigger">
        <span class="menu-icon">
          <svg
            xmlns="http://www.w3.org/2000/svg"
            width="24"
            height="24"
            viewBox="0 0 512 512"
          >
            <path
              d="M64,384H448V341.33H64Zm0-106.67H448V234.67H64ZM64,128v42.67H448V128Z"
            />
          </svg>
        </span>
      </label>
      <a id="mode">
        <svg
          class="mode-sunny"
          xmlns="http://www.w3.org/2000/svg"
          width="24"
          height="24"
          viewBox="0 0 512 512"
        >
          <title>LIGHT</title>
          <line
            x1="256"
            y1="48"
            x2="256"
            y2="96"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="256"
            y1="416"
            x2="256"
            y2="464"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="403.08"
            y1="108.92"
            x2="369.14"
            y2="142.86"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="142.86"
            y1="369.14"
            x2="108.92"
            y2="403.08"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="464"
            y1="256"
            x2="416"
            y2="256"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="96"
            y1="256"
            x2="48"
            y2="256"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="403.08"
            y1="403.08"
            x2="369.14"
            y2="369.14"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="142.86"
            y1="142.86"
            x2="108.92"
            y2="108.92"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <circle
            cx="256"
            cy="256"
            r="80"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
        </svg>
        <svg
          class="mode-moon"
          xmlns="http://www.w3.org/2000/svg"
          width="24"
          height="24"
          viewBox="0 0 512 512"
        >
          <title>DARK</title>
          <line
            x1="256"
            y1="48"
            x2="256"
            y2="96"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="256"
            y1="416"
            x2="256"
            y2="464"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="403.08"
            y1="108.92"
            x2="369.14"
            y2="142.86"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="142.86"
            y1="369.14"
            x2="108.92"
            y2="403.08"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="464"
            y1="256"
            x2="416"
            y2="256"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="96"
            y1="256"
            x2="48"
            y2="256"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="403.08"
            y1="403.08"
            x2="369.14"
            y2="369.14"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <line
            x1="142.86"
            y1="142.86"
            x2="108.92"
            y2="108.92"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
          <circle
            cx="256"
            cy="256"
            r="80"
            style="stroke-linecap:round;stroke-miterlimit:10;stroke-width:32px"
          />
        </svg>
      </a>
      <div class="trigger">
        <div class="trigger-container">
          
            
            
            
            
            
            
            <a class="menu-link " href="/"> Home</a>
          
            
            
            
            
            
            
            <a class="menu-link " href="/about/"> About me</a>
          
            
            
            
            
            
            
            <a class="menu-link " href="/publications/"> Publications</a>
          
            
            
            
            
            
            
            <a class="menu-link " href="/projects/"> Projects</a>
          
            
            
            
            
            
            
            <a class="menu-link " href="/cv/mycv.pdf"> CV</a>
          
            
            
            
            
            
            
            <a class="menu-link " target="_blank" rel="noopener" href="https://github.com/minsilu"> GitHub</a>
          
            
            
            
            
            
            
            <a class="menu-link " href="/blog/"> Blog</a>
          
            
            
            
            
            
            
            <a class="menu-link " href="/contact/"> Contact</a>
          
        </div>
      </div>
    </nav>
  </div>
  
<div class="wrapper">
  <header class="header">
    <h1 class="header-title center">PPI Network Guided Driver Target Discovery</h1>
  </header>
  <main class="page-content" aria-label="Content">
      <div class="post">
        <p>Wenda Chu, Minsi Lu, Botian Wang, Sihang Zeng, Shiyu Zhao</p>
<p>Repo: <a target="_blank" rel="noopener" href="https://github.com/minsilu/PPI-Guided-Driver-Target-Discovery">PPI Network Guided Driver Target Discovery</a></p>
<h2 id="Introduction"><a href="#Introduction" class="headerlink" title="Introduction"></a>Introduction</h2><p>With the development of high-throughput biological sequencing techniques, single-cell data are becoming an important data modality in the field of bio-informatics and data mining. In this project, we focus on the problem of driver target discovery using single-cell RNA-seq data. The problem of target discovery aims to mine possible gene targets for a specific disease, and is a fundamental task for drug design and pathway discovery. In this project, the input of the target discovery framework is a given list of known disease targets and the single-cell RNA-seq data matrix, and the output is a list of predicted new driver targets. Additionally, the protein-protein interaction (PPI) data are leveraged in our framework to model the relationships between genes, which proves to be helpful for the discovery of potential targets.</p>
<h2 id="Method"><a href="#Method" class="headerlink" title="Method"></a>Method</h2><p>Our key motivation is that genes in the same pathway should be close together in the embedding space to find more potential targets near known targets in the embedding space. Note that the PPI graph contains interactions between proteins (genes), including shared function, binding, activation, and inhibition, which incorporates important pathway information for various diseases. Hence, we choose to use a PPI network to add constraints on the gene embedding space.</p>
<h3 id="Framework"><a href="#Framework" class="headerlink" title="Framework"></a>Framework</h3><p>Our overall framework is presented in the figure below. Firstly, we use sc-ETM as the <strong>gene feature extractor</strong>. The training of sc-ETM is <strong>regularized</strong> by an extracted PPI network. Then, the <strong>gene embeddings</strong> are fed into the <strong>target score regression head</strong>, which outputs a score for each possible target. After sorting the results and selecting the top k targets, we obtain a list of the predicted gene targets.<br><img src="/projects/ppi/f2.png" alt="f2"></p>
<h3 id="PPI-Network-Construction"><a href="#PPI-Network-Construction" class="headerlink" title="PPI Network Construction"></a>PPI Network Construction</h3><p>We download the PPI network of homo sapiens from <a target="_blank" rel="noopener" href="https://string-db.org/cgi/download?sessionId=bDGKTiiGcsNX">STRING 11.5 database</a>. After filtering with genes in the RNA-seq dataset, we obtain 6184316 PPI links among 12015 genes and construct an undirected graph as our PPI network. Each PPI link is defined as $(u, e, v)$, where $u, v$ are two genes (proteins) and $e$ is the confidence score of the interaction between $u, v$.</p>
<h3 id="Regularization-on-PPI"><a href="#Regularization-on-PPI" class="headerlink" title="Regularization on PPI"></a>Regularization on PPI</h3><p>Following the intuition that genes with strong biological correlation should be mapped to similar vectors in the embedding space, we introduce a regularization loss to inject biological information into the gene embedding. Let $G &#x3D; (V, E)$ be the graph of a PPI network, each edge $e &#x3D; (u, v)$ of which is assigned with weight $w_e$ representing the strength of biological correlation between two genes. Our regularization loss $L_{reg}$ computes a weighted sum of squared $\ell_2$ distances between all gene pairs. The gene pairs with stronger correlation according to the PPI network have a larger weight we, so they are guided to be mapped closely in the embedding space.<br><img src="/projects/ppi/e1.png" alt="e1"></p>
<p>where $\rho_u, \rho_v \in R^d$ are the gene embedding of $u$ and $v$, respectively. We combine the regularization loss with the original ELBO loss function from sc-ETM to train our model.<br><img src="/projects/ppi/e2.png" alt="e2"><br>where $λ_{reg}$ is a hyper-parameter, chosen to be 1e-2 in our experiments.</p>
<h2 id="Result"><a href="#Result" class="headerlink" title="Result"></a>Result</h2><h3 id="Dataset"><a href="#Dataset" class="headerlink" title="Dataset"></a>Dataset</h3><p>sc-RNA seq: Glioma transcriptional Cells (GBM, IDH-MUT)</p>
<ul>
<li>Glioma is a type of tumor that occurs in brain and spinal cord</li>
<li>Single-cell data: 948 cells * 12916 genes</li>
<li>RNA expression matrix provided by Chaligne et al., which comprises of data from the IDH-mutant and IDH-wildtype glioblastoma.To our best knowledge, there lacks mutant-specific target data for glioma in existing databases, so we only use the IDH-wildtype cells for the target discovery task.</li>
</ul>
<p>Target Dataset: Open Targets Dataset</p>
<ul>
<li>7751 Glioma targets with target scores</li>
<li>4699 overlapping genes with GBM<br>Train &#x2F; Valid &#x2F; Test : 8&#x2F;1&#x2F;1</li>
</ul>
<h3 id="Clustering-Performance"><a href="#Clustering-Performance" class="headerlink" title="Clustering Performance"></a>Clustering Performance</h3><p>We evaluate the performance of clustering on the embeddings to verify the effectiveness of PPI network guidance. We follow the evaluation metric of the original paper and use Leiden to cluster cells by their cell embeddings. During the clustering, we tried multiple resolution values and reported the result with the highest ARI and NMI for each method. The results are as follows:<br><img src="/projects/ppi/t1.png" alt="t1"></p>
<p>We also visualize the clustering results in Figure 2. The figure on the left-hand side is the clustering result with the highest ARI and NMI(the number of labels is not designated). As illustrated, the best clustering result without PPI guidance corresponds to two labels, while that with PPI guidance corresponds to three labels, which is the true number of labels.</p>
<p>Besides, the figure on the right-hand side is the clustering result given the true number of labels. In the original model without PPI guidance, the 0.0(blue) cluster and 1.0(yellow) cluster are close to each other and are not separated well, while in the model with PPI guidance, all of the three clusters are well separated: 0.0(blue) cluster on the left top, 1.0(yellow) cluster on the right bottom, and 2.0(green) cluster in the middle.<br><img src="/projects/ppi/f3.png" alt="f3"></p>
<h3 id="Driver-Target-Discovery-Performance"><a href="#Driver-Target-Discovery-Performance" class="headerlink" title="Driver Target Discovery Performance"></a>Driver Target Discovery Performance</h3><p>We evaluate our potential targets on the test set, which is unseen during training and mimics the actual target discovery pipeline. We adopt two metrics as evaluation, namely, root mean square error (RMSE) and Spearman’s rank correlation (Spearman’s r). We use 5 different random seeds for each model to avoid randomness. The results are shown in the Figure, where our&#x2F;baseline method refers to using the gene embeddings with&#x2F;without PPI regularization.<br><img src="/projects/ppi/f4.png" alt="f4"><br>It is shown that our method has lower RMSE and higher Spearman’s r, with much lower variance. We then conduct a two-sample t-test and obtain the p-value of 1.458e-5 and 0.0037 respectively for RMSE and Spearman’s r, which indicates that there are significant improvements of the performance.</p>
<p>After predicting these genes, how do we determine if these genes are reliable drug targets? From a biological point of view, it is very important to determine whether the protein is expressed differently in tumor tissues than in normal cells, whether some pathways can be identified to prove that it plays a key role in the disease process, and whether there are successful results of functional validation of the genes. We perform pathway analysis to the 10 candidate genes with the highest regression score and we collected some existing experiments (e.g. gene knockout) that verify the functions of the genes. Our findings are organized in the table below.<br><img src="/projects/ppi/t2.png" alt="t2"></p>

       </div>
         <div class="dis">
    
     

  </div>
  </main>
</div>
  <footer class="footer">
    <small class="footer_copyright">
      <div id="bottom-inner">
        © 2021 Site by Minsi Lu.
      </div>
    </small>
  </footer>
  
  

    
      <script src="/js/main.js"></script>
    
      <script src="/js/custom-filter.js"></script>
    
  
  <script>
    window.FPConfig = {
      delay: 0,
      ignoreKeywords: [],
      maxRPS: 3,
      hoverDelay: 50,
    };
  </script>
  <script defer src="https://cdn.jsdelivr.net/gh/gijo-varghese/flying-pages@2.1.2/flying-pages.min.js"></script>



  
</body>
</html>
